dns/zone.py

```diff
@@ -500,16 +501,27 @@ def to_file(self, f, sorted=True, relativize=True, nl=None):
 
         str_type = string_types
 
-        if nl is None:
-            opts = 'wb'
-        else:
-            opts = 'wb'
-
         if isinstance(f, str_type):
-            f = open(f, opts)
+            f = open(f, 'wb')
             want_close = True
         else:
             want_close = False
+
+        # must be in this way, f.encoding may contain None, or even attribute
+        # may not be there
+        file_enc = getattr(f, 'encoding', None)
+        if file_enc is None:
+            file_enc = 'utf-8'
```

Comment:
In [3]: f = open('testfile', 'wb')
In [4]: f.write(u'čččč'.encode('utf-8'))
Out[4]: 8
In [5]: f.write(u'čččč'.encode('utf-32'))
Out[5]: 20
In [6]: f.write(u'čččč'.encode('ascii'))
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-5-32d0706eb9ad> in <module>()
----> 1 f.write(u'čččč'.encode('ascii'))
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-3: ordinal not in range(128)
In [7]: f.flush()
In [8]: f.close()

If you open that file in gedit for example, you will see only 'čččč', so it is depending on expected file encoding (on linux utf-8 usually)

I'm not sure if 
self[n].to_text()
may return a non-ascii character, so thats why I played safe with utf-8, because ASCII characters will be same, and other characters will be dropped to file as utf-8